src/_pytest/assertion/util.py

```diff
@@ -501,10 +501,19 @@ def _compare_eq_dict(
     if diff:
         explanation += ["Differing items:"]
         for k in diff:
+            left_val = left[k]
+            right_val = right[k]
+            if issequence(left_val) and issequence(right_val):
```

Comment:
I'm not sure I like this approach, given this is really only specific for sequences inside dictionaries... this is not general because really we could have anything in there, for example other dicts or dataclasses, each with their own custom diff code... 🤔 

The core of the original issue was the fact that it was *truncating* the diff. One idea is to not truncate anything when computing the diff, however truncation is there for a reason, to avoid dumping MBs of text in case of large data structures...

I'm a bit -1 on this solution, given it only fixes a very specific use case and makes the code slightly worse, but I don't have a definitive suggestion either.

Reply:
Yeah I initially thought we could take all the cases in `_compare_eq_any` and use those to create the new diffs but they seemed too long for cases outside of sequences due to their multi-line diffs, e.g.:
```
E       assert {1, 2} == {1, 3}
E         
E         Extra items in the left set:
E         2
E         Extra items in the right set:
E         3
```
Besides this though, I figured sequence comparisons would be the most likely use case, and it'd be worth having it there despite the bit of complexity. I could be wrong though!

Reply:
An alternative method might be to figure out a truncated diff for the other data types such as a simple `Differing items` for each instead of their specific diffs, but this seems overkill for just comparisons within dictionaries